### DISCUSSION TITLE
Facing the issue of being unable to use Ktlint old version (0.45.2) for KDocs.

### DISCUSSION THREAD

**Discussion by jainv4156** [ktlint/ktlint#2870] [Q&A]
I am using Ktlint version (0.45.2) and Kotlin (1.6.10). I want to use Ktlint for KDocs, but Ktlint is unable to detect any errors in the KDocs.
This is how I am using Ktlint in my project:

# script 

java -jar $jar_file_path --android --verbose --experimental --debug app/src/main/java/../ActivityRouter.kt

# editorconfig file 
[*.{kt,kts}]

indent_size = 2
ktlint_kdoc_style = true

# debug info 

0.45.2
21:46:28.389 [main] DEBUG com.pinterest.ktlint.KtlintCommandLine - Discovered reporter with "baseline" id.
21:46:28.390 [main] DEBUG com.pinterest.ktlint.KtlintCommandLine - Discovered reporter with "checkstyle" id.
21:46:28.390 [main] DEBUG com.pinterest.ktlint.KtlintCommandLine - Discovered reporter with "json" id.
21:46:28.390 [main] DEBUG com.pinterest.ktlint.KtlintCommandLine - Discovered reporter with "html" id.
21:46:28.390 [main] DEBUG com.pinterest.ktlint.KtlintCommandLine - Discovered reporter with "plain" id.
21:46:28.390 [main] DEBUG com.pinterest.ktlint.KtlintCommandLine - Discovered reporter with "sarif" id.
21:46:28.390 [main] DEBUG com.pinterest.ktlint.KtlintCommandLine - Initializing "plain" reporter with {verbose=true, color=false, color_name=DARK_GRAY}
[DEBUG] Rules will be executed in order below (unless disabled):
           - standard:filename, 
           - standard:final-newline, 
           - standard:wrapping, 
           - standard:chain-wrapping, 
           - standard:colon-spacing, 
           - standard:comma-spacing, 
           - standard:comment-spacing, 
           - standard:curly-spacing, 
           - standard:dot-spacing, 
           - standard:import-ordering, 
           - standard:keyword-spacing, 
           - standard:modifier-order, 
           - standard:no-blank-line-before-rbrace, 
           - standard:no-consecutive-blank-lines, 
           - standard:no-empty-class-body, 
           - standard:no-line-break-after-else, 
           - standard:no-line-break-before-assignment, 
           - standard:no-multi-spaces, 
           - standard:no-semi, 
           - standard:no-trailing-spaces, 
           - standard:no-unit-return, 
           - standard:no-unused-imports, 
           - standard:no-wildcard-imports, 
           - standard:op-spacing, 
           - standard:parameter-list-wrapping, 
           - standard:paren-spacing, 
           - standard:range-spacing, 
           - standard:string-template, 
           - experimental:annotation, 
           - experimental:annotation-spacing, 
           - experimental:argument-list-wrapping, 
           - experimental:comment-wrapping, 
           - experimental:discouraged-comment-location, 
           - experimental:double-colon-spacing, 
           - experimental:enum-entry-name-case, 
           - experimental:fun-keyword-spacing, 
           - experimental:function-type-reference-spacing, 
           - experimental:kdoc-wrapping, 
           - experimental:modifier-list-spacing, 
           - experimental:multiline-if-else, 
           - experimental:no-empty-first-line-in-method-block, 
           - experimental:package-name, 
           - experimental:spacing-around-angle-brackets, 
           - experimental:spacing-between-declarations-with-annotations, 
           - experimental:spacing-between-declarations-with-comments, 
           - experimental:type-argument-list-spacing, 
           - experimental:type-parameter-list-spacing, 
           - experimental:unary-op-spacing, 
           - experimental:unnecessary-parentheses-before-trailing-lambda, 
           - standard:indent, 
           - experimental:block-comment-initial-star-alignment, 
           - experimental:trailing-comma, 
           - standard:max-line-length
           
my output does not found any error related to kdoc , can anyone help me

**Comment by paul-dingemans** [comment]
Indeed this is a very old version of ktlint that you are using. I hope you can understand that no support is given on that anymore. Try to upgrade to a newer version (latest version at this moment is `1.4.1`) and check whether KDocs are still not checked.
Also it will help if you provide an example file, and the expected output.

**Reply by jainv4156** [reply]
i am using ktlint version 1.4.0 , with kotlint 1.9.0 , can you please tell me how can i use ktlint for kdoc 

# Output 
********************************
Checking code formatting
********************************
ktlint version 1.4.0
Desktop/Newfolder/kotlinProject/Ktlinttesting/app/src/main/java/com/example/ktlinttesting/LintTesting.kt:12:10: Function body should be replaced with body expression (standard:function-expression-body)
10:52:33.962 [main] WARN com.pinterest.ktlint.cli.internal.KtlintCommandLine -- Lint has found errors than can be autocorrected using 'ktlint --format'

Summary error count (descending) by rule:
  standard:function-expression-body: 1
********************************

# sample file 
package com.example.ktlinttesting

/** sjadfljdljaklsfdj
 * asdkas;dfjklas
 * asdkjalsd
 * */
class LintTesting {
  /**  sum the a and b*/
  fun sum(
    a: Int,
    b: Int,
  ): Int {
    return a + b
  }

  val sum = sum(1, 3)
}
# script
java -jar $jar_file_path  app/src/main/java/com/example/ktlinttesting/LintTesting.kt

# editorconfig file
[*.{kt,kts}]

indent_size = 2

**Reply by paul-dingemans** [reply]
Your code sample above is not formatted properly. Please put it in a code block. Also, you don't make clear what you expect in the output. I can start guessing, but I rather would like to know your expectations.
